Casks/minecraft-server.rb

```diff
@@ -4,10 +4,18 @@
 
   url "https://launcher.mojang.com/v#{version.major}/objects/#{version.after_comma}/server.jar",
       verified: "launcher.mojang.com/"
-  appcast "https://minecraft.net/en-us/download/server/"
   name "Minecraft Server"
+  desc "Run a Minecraft multiplayer server"
   homepage "https://minecraft.net/"
 
+  livecheck do
+    url "https://minecraft.net/en-us/download/server/"
+    strategy :page_match do |page|
+      match = page.match(%r{href=.*?/objects/(\h+)/server\.jar[^>]*>minecraft[_-]server[._-]v?(\d+(?:\.\d+)*).jar}i)
+      "#{match[2]},#{match[1]}"
```

Comment:
```suggestion
    url "https://minecraft.net/en-us/download/server/"
    strategy :page_match do |page|
      page.scan(%r{href=.*?/objects/(\h+)/server\.jar[^>]*>minecraft[_-]server[._-]v?(\d+(?:\.\d+)*)\.jar}i)
          .map do |match|
        "#{match[1]},#{match[0]}"
```

The `minecraft.net` URL redirects to `www.minecraft.net`, so we can update the URL to avoid the redirection.

In general, I would say it's better to always use `page.scan(regex).map do |match|` instead of `match = page.match(regex)`, as `#scan` will match all occurrences of a regex but `#match` only matches the first occurrence. `PageMatch` uses `#scan` internally and we simply leave it up to livecheck to go through the matched versions and determine which is newest (using `Version` comparison).

Basically, `#scan` will continue to work properly if the page is updated to list more than one version whereas `#match` could potentially return an incorrect result. It's a small change but it can help reduce the maintenance burden over time.

Reply:
I usually check the `homepage`, `stable`/`url`, etc. when I find a redirection somewhere in a formula/cask and update any URLs that are redirecting. In this case, the `homepage` also redirects to `https://www.minecraft.net/`, so we may want to update that while we're at it.

Reply:
I'm happy to do whatever you prefer but the reason I split it up was because `brew style` complained that the line was too long.

Reply:
It may be easy to overlook but I addressed the line length issue in the suggestion above by having `.map` on a separate line (indented to align with `.scan`). This is what the current `PageMatch` `strategy` block example in the `brew livecheck` documentation (following the homebrew/cask style) does to work around longer regexes that go past the line length limit.

Reply:
Ah, gotcha 👍